src/array.jl

```diff
@@ -212,7 +212,7 @@ Base.convert(::Type{T}, x::T) where T <: CuArray = x
 function Base._reshape(parent::CuArray, dims::Dims)
   n = length(parent)
   prod(dims) == n || throw(DimensionMismatch("parent has $n elements, which is incompatible with size $dims"))
-  return CuArray{eltype(parent),length(dims)}(parent.ptr, dims, parent)
+  return CuArray{eltype(parent),length(dims)}(pointer(parent), dims, parent)
```

Comment:
I guess this doesn't need one assuming all other uses of `CuArray` have the object itself preserved....